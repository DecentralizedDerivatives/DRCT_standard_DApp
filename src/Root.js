import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const middlewares = [thunk];
let composeEnhancer = compose;

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger());
  composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const Root = ({ children, initialState = {} }) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...middlewares))
  );

  return <Provider store={store}>{children}</Provider>;
};

export default Root;
