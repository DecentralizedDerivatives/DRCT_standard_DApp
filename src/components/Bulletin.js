import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import OrderBook from './OrderBook';
import RecentTrades from './RecentTrades';
import ContractDetails from './ContractDetails';
import MyPositionsBulletin from './MyPositionsBulletin';
import PriceChart from './PriceChart';
import ListFormContainer from './ListFormContainer';
import ApprovalFormContainer from './ApprovalFormContainer';

import {
  getUserAccount,
  getUserTokenPositions,
  getUserOrders,
  getUserPositions,
} from '../actions/userActions';
import {
  getOrderBook,
  getRecentTrades,
  getContractDetails
} from '../actions/contractActions';
import requireConnection from './requireConnection';

// Use named export for unconnected component for testing
export class Bulletin extends Component {
  constructor(props) {
    super(props);

    this.state = {
      detailsOpen: false,
      resultsMessage: '',
      token: '',
      balance: 0,
    };
    this.handleRowClick = this.handleRowClick.bind(this)
    this.handleList = this.handleList.bind(this)
  }

  async componentDidMount() {
    await this.props.getUserAccount();
    if (!this.props.userAccount) { return };
    this.props.getOrderBook();
    this.props.getRecentTrades();
    this.props.getUserPositions(this.props.userAccount);
    this.props.getUserTokenPositions(this.props.userAccount);
    this.props.getUserOrders(this.props.userAccount);
    this.orderBookInterval = setInterval(() => this.props.getOrderBook(this.props.userAccount, true), 30000);
    this.positionsInterval = setInterval(() => this.props.getUserPositions(this.props.userAccount, true), 30000);
    this.recentTradesInterval = setInterval(() => this.props.getRecentTrades(this.props.userAccount, true), 30000);
  }
  componentWillUnmount() {
    clearInterval(this.orderBookInterval);
    clearInterval(this.positionsInterval);
    clearInterval(this.recentTradesInterval);
  }

  handleRowClick = async (tokenAddress, symbol, date, e) => {
    e.preventDefault();
    await this.props.getContractDetails(symbol, date);
    this.setState({
      detailsOpen: true
    });
  };

  refreshPage = (silentMode) => {
    this.props.getOrderBook(this.props.userAccount, silentMode)
    this.props.getRecentTrades(this.props.userAccount, silentMode)
  }
  renderContractDetails = () => {
    return this.state.detailsOpen ? (
      <ContractDetails
        close={this.closeContractDetails.bind(this)} />
    ) : null;
  };
  closeContractDetails = () => {
    this.setState({
      detailsOpen: false
    });
  };

  handleList = (token,balance) => {
    alert('yo')
    this.setState({
      listOrderOpen: true,
      token: token,
      balance: balance,
    });
  }

  render() {
    return (
      <div id="bulletin">
        <OrderBook onRowClick={this.handleRowClick} refreshPage={this.refreshPage} />
        <MyPositionsBulletin handleList={this.handleList} refreshPage={this.refreshPage}/>
        
        <div className="price-chart">
          {this.props.userAccount ? <PriceChart /> : ''}
        </div>
        <RecentTrades onRowClick={this.handleRowClick} />

        {this.renderContractDetails()}
      </div>
    );
  }
}
Bulletin.propTypes = {
  getUserAccount: PropTypes.func.isRequired,
  getUserPositions: PropTypes.func.isRequired,
  getOrderBook: PropTypes.func.isRequired,
  getRecentTrades: PropTypes.func.isRequired,
  getContractDetails: PropTypes.func.isRequired,
  getUserTokenPositions: PropTypes.func.isRequired,
  getUserOrders: PropTypes.func.isRequired,
  orderId: PropTypes.string,
  userAccount: PropTypes.string
};

const mapStateToProps = state => ({
  userAccount: state.user.userAccount
});

export default connect(
  mapStateToProps,
  {
    getUserAccount,
    getOrderBook,
    getRecentTrades,
    getContractDetails,
    getUserTokenPositions,
    getUserOrders,
    getUserPositions,
  }
)(requireConnection(Bulletin));
