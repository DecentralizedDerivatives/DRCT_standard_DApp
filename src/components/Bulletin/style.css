#bulletin{
    text-align: center;
    vertical-align: top;
}
.bulletin-item{
    margin:25px 0;
}
.bulletin-item-left{
    display: inline-block;
    width: 55vw;
    float:left;
    margin-left: 3em;
}
.bulletin-item-left .Table-root-301{
    width:100%;
}
.bulletin-item-right{
    display:inline-block;
    margin-right: 3em;
    float:right;
    width:30vw;
    text-align: center;
}
.bulletin-order-btns-wrapper{
    list-style:none;
    margin:0;
    padding:0;
    width:90%;
    display:inline-block;
}
.bulletin-order-btns li{
    width:100%;
}
.bulletin-order-btn{
    background-color:transparent;
    color: white;
    width: 300px;
    height: 45px;
    border-bottom: 1px solid #27597E!important;
    border-radius: 5px;

}
.bulletin-order-btn-txt{
    color: black;
    font-size:18px;
    font-weight:500;
    letter-spacing: 1px;
}