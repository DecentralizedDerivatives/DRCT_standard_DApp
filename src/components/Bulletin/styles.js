const button = {
  height: '80px',
  borderRight: '1px solid #27597E',
  padding: '0px 30px',
  borderRadius: '0px',
};
const text = theme => ({
  color: theme.colors.white,
  fontSize: theme.fonts.size.header,
  fontWeight: theme.fonts.weight.medium,
  letterSpacing: '1px',
});

export default theme => ({
  item: {
    width: '90%',
    marginTop: '25px',
  },
  container: {
    marginBottom: '25px',
  },
  button: {
    backgroundColor:theme.colors.white,
    border: '1px solid #27597E',
    color: 'white',
    width: 300,
    height: 45,
    borderBottom:'1px solid #27597E',
    borderWidth: 0,
    borderRadius: 5,
    ...text(theme),
  },
  buttonActive: {
    ...button,
    ...text(theme),
    backgroundColor: theme.colors.white,
    '&:hover': {
      backgroundColor: theme.colors.white,
    },
  },
});
