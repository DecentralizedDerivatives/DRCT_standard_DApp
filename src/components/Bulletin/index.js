import Bulletin from './Bulletin';

export default Bulletin;
