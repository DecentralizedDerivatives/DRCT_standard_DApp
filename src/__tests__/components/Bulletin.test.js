import React from 'react';
import { shallow } from 'enzyme';
import { initStore } from '../../Root';
import Bulletin from '../../components/Bulletin';

describe('<Bulletin />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <Bulletin store={initStore()} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
