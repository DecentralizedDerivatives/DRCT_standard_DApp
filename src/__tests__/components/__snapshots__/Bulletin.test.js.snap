// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bulletin /> handles details 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bulletin
    dispatch={[Function]}
    getContractDetails={[Function]}
    getOrderBook={[Function]}
    getRecentTrades={[Function]}
    getUserAccount={[Function]}
    getUserOrders={[Function]}
    getUserTokenPositions={[Function]}
    metamask={true}
    network={4}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount="0xd8e09fbab17f30e1C2b2Ae85A26EA1F3F31267ef"
    verified={true}
    whiteListed={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(OrderBook)
          onRowClick={[Function]}
          refreshPage={[Function]}
        />,
        <div
          className="order-buttons"
        >
          <Connect(List)
            refreshPage={[Function]}
          />
        </div>,
        <div
          className="price-chart"
        >
          <Connect(PriceChart) />
        </div>,
        <Connect(RecentTrades)
          onRowClick={[Function]}
        />,
        <Connect(ContractDetails)
          close={[Function]}
        />,
      ],
      "id": "bulletin",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
          "refreshPage": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(List)
            refreshPage={[Function]}
          />,
          "className": "order-buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(PriceChart) />,
          "className": "price-chart",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "close": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(OrderBook)
            onRowClick={[Function]}
            refreshPage={[Function]}
          />,
          <div
            className="order-buttons"
          >
            <Connect(List)
              refreshPage={[Function]}
            />
          </div>,
          <div
            className="price-chart"
          >
            <Connect(PriceChart) />
          </div>,
          <Connect(RecentTrades)
            onRowClick={[Function]}
          />,
          <Connect(ContractDetails)
            close={[Function]}
          />,
        ],
        "id": "bulletin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(List)
              refreshPage={[Function]}
            />,
            "className": "order-buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "refreshPage": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(PriceChart) />,
            "className": "price-chart",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "close": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Bulletin /> handles details 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bulletin
    dispatch={[Function]}
    getContractDetails={[Function]}
    getOrderBook={[Function]}
    getRecentTrades={[Function]}
    getUserAccount={[Function]}
    getUserOrders={[Function]}
    getUserTokenPositions={[Function]}
    metamask={true}
    network={4}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount="0xd8e09fbab17f30e1C2b2Ae85A26EA1F3F31267ef"
    verified={true}
    whiteListed={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(OrderBook)
          onRowClick={[Function]}
          refreshPage={[Function]}
        />,
        <div
          className="order-buttons"
        >
          <Connect(List)
            refreshPage={[Function]}
          />
        </div>,
        <div
          className="price-chart"
        >
          <Connect(PriceChart) />
        </div>,
        <Connect(RecentTrades)
          onRowClick={[Function]}
        />,
        null,
      ],
      "id": "bulletin",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
          "refreshPage": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(List)
            refreshPage={[Function]}
          />,
          "className": "order-buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(PriceChart) />,
          "className": "price-chart",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(OrderBook)
            onRowClick={[Function]}
            refreshPage={[Function]}
          />,
          <div
            className="order-buttons"
          >
            <Connect(List)
              refreshPage={[Function]}
            />
          </div>,
          <div
            className="price-chart"
          >
            <Connect(PriceChart) />
          </div>,
          <Connect(RecentTrades)
            onRowClick={[Function]}
          />,
          null,
        ],
        "id": "bulletin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(List)
              refreshPage={[Function]}
            />,
            "className": "order-buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "refreshPage": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(PriceChart) />,
            "className": "price-chart",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Bulletin /> renders empty component and unmounts 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bulletin
    dispatch={[Function]}
    getContractDetails={[Function]}
    getOrderBook={[Function]}
    getRecentTrades={[Function]}
    getUserAccount={[Function]}
    getUserOrders={[Function]}
    getUserTokenPositions={[Function]}
    metamask={false}
    network={0}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount=""
    verified={false}
    whiteListed={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(OrderBook)
          onRowClick={[Function]}
          refreshPage={[Function]}
        />,
        <div
          className="order-buttons"
        >
          <Connect(List)
            refreshPage={[Function]}
          />
        </div>,
        <div
          className="price-chart"
        >
          
        </div>,
        <Connect(RecentTrades)
          onRowClick={[Function]}
        />,
        null,
      ],
      "id": "bulletin",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
          "refreshPage": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(List)
            refreshPage={[Function]}
          />,
          "className": "order-buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "price-chart",
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(OrderBook)
            onRowClick={[Function]}
            refreshPage={[Function]}
          />,
          <div
            className="order-buttons"
          >
            <Connect(List)
              refreshPage={[Function]}
            />
          </div>,
          <div
            className="price-chart"
          >
            
          </div>,
          <Connect(RecentTrades)
            onRowClick={[Function]}
          />,
          null,
        ],
        "id": "bulletin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(List)
              refreshPage={[Function]}
            />,
            "className": "order-buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "refreshPage": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "price-chart",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Bulletin /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bulletin
    dispatch={[Function]}
    getContractDetails={[Function]}
    getOrderBook={[Function]}
    getRecentTrades={[Function]}
    getUserAccount={[Function]}
    getUserOrders={[Function]}
    getUserTokenPositions={[Function]}
    metamask={true}
    network={4}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userAccount="0xd8e09fbab17f30e1C2b2Ae85A26EA1F3F31267ef"
    verified={true}
    whiteListed={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(OrderBook)
          onRowClick={[Function]}
          refreshPage={[Function]}
        />,
        <div
          className="order-buttons"
        >
          <Connect(List)
            refreshPage={[Function]}
          />
        </div>,
        <div
          className="price-chart"
        >
          <Connect(PriceChart) />
        </div>,
        <Connect(RecentTrades)
          onRowClick={[Function]}
        />,
        null,
      ],
      "id": "bulletin",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
          "refreshPage": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(List)
            refreshPage={[Function]}
          />,
          "className": "order-buttons",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(PriceChart) />,
          "className": "price-chart",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onRowClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(OrderBook)
            onRowClick={[Function]}
            refreshPage={[Function]}
          />,
          <div
            className="order-buttons"
          >
            <Connect(List)
              refreshPage={[Function]}
            />
          </div>,
          <div
            className="price-chart"
          >
            <Connect(PriceChart) />
          </div>,
          <Connect(RecentTrades)
            onRowClick={[Function]}
          />,
          null,
        ],
        "id": "bulletin",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
            "refreshPage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(List)
              refreshPage={[Function]}
            />,
            "className": "order-buttons",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "refreshPage": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(PriceChart) />,
            "className": "price-chart",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onRowClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
